### ISSUE TITLE
pivot

### ISSUE THREAD

**Issue by isaiahcampusano** [isaiahcampusano/garvis#1]
<html>
<body>
<html><head></head><body><p>Yeah — the OpenAI roles we were looking at are mostly <strong>not “JPMC analyst program” entry-level roles</strong>.</p><p>OpenAI does have true early-career hiring: its Emerging Talent page explicitly says it offers full-time roles in research, applied engineering, and product for people with <strong>0–3 years of experience</strong>, including recent grads and self-taught candidates. (<a href="https://openai.com/careers/emerging-talent/?utm_source=chatgpt.com" title="OpenAI">OpenAI</a>) But the specific <strong>Full-Stack Engineer, ChatGPT Education &amp; Learning</strong> posting is not labeled as an early-career program role. It reads like they want someone who can already operate as an independent product engineer: own features end to end, build across frontend and backend, ship polished experiences, work with React/TypeScript plus Python/Node services, integrate AI capabilities, and handle ambiguity without much scaffolding. (<a href="https://openai.com/careers/full-stack-engineer-chatgpt-education-and-learning-san-francisco/?utm_source=chatgpt.com" title="Full-Stack Engineer, ChatGPT Education &amp; Learning | OpenAI">OpenAI</a>)</p><p>So I’d think of the comparison like this:</p>
JPMC analyst-style role | OpenAI Education full-stack role
-- | --
“We’ll train a strong new grad” | “Show us you can already ship”
Potential-heavy | Evidence-heavy
Structured ramp | High ownership quickly
Coursework + internship pedigree matter a lot | Portfolio + production-quality work matter a lot

<p>For the Education role specifically, I would <strong>not wait until you look like some 10-year senior engineer</strong>. The current posting doesn’t specify a hard years-of-experience requirement. What matters is reaching the point where your GitHub makes someone believe, “This person could own a real feature.” (<a href="https://openai.com/careers/full-stack-engineer-chatgpt-education-and-learning-san-francisco/?utm_source=chatgpt.com" title="Full-Stack Engineer, ChatGPT Education &amp; Learning | OpenAI">OpenAI</a>)</p><p>For you, that means moving from <strong>lots of interesting prototypes</strong> to roughly <strong>2–4 extremely convincing full-stack projects</strong>. The portfolio should communicate these capabilities:</p><ul><li><p>You can build a real React/TypeScript frontend rather than just static demos.</p></li><li><p>You can design a backend/API in Python or Node and understand the data flow.</p></li><li><p>You can persist user state with a database and handle authentication or accounts where appropriate.</p></li><li><p>You can integrate an LLM or multimodal model for a reason deeper than “I called an API.”</p></li><li><p>You understand loading/error states, retries, validation, basic security, and edge cases.</p></li><li><p>You can test meaningful logic and explain architectural decisions.</p></li><li><p>At least one project is actually deployed and usable.</p></li><li><p>Most importantly for this team, you can demonstrate <strong>product judgment around learning</strong>: what problem does the learner have, why does the interaction work, and how did you evaluate whether it helped?</p></li></ul><p>That last one is where you could build a very differentiated portfolio.</p><p>Your current Fourier/softmax-style educational work is pointing in the right direction, but the OpenAI version of that portfolio shouldn't just be:</p><blockquote><p>“Here is an interactive visualization of softmax.”</p></blockquote><p>It should eventually resemble something like:</p><blockquote><p>A learner says, “I don’t understand why increasing one logit affects every other probability.” The product identifies the misconception, generates an interactive visualization, lets the learner manipulate logits, adapts the explanation based on their actions, remembers where they struggled, and tests whether they actually understand the concept afterward.</p></blockquote><p>Now you have <strong>frontend + backend + AI + interaction design + learning science + user state + evaluation</strong> in one coherent product. That maps much more directly to what the Learning team says it builds: end-to-end learning experiences, multimodal interfaces, scalable backend services, experimentation, user feedback, and AI-powered educational products. (<a href="https://openai.com/careers/full-stack-engineer-chatgpt-education-and-learning-san-francisco/?utm_source=chatgpt.com" title="Full-Stack Engineer, ChatGPT Education &amp; Learning | OpenAI">OpenAI</a>)</p><h3>The level I would target</h3><p>I’d use this mental ladder:</p><p><strong>Level 1 — Prototype builder</strong><br>“Given an idea and AI coding tools, I can get something working.”</p><p>You’re already around here across multiple projects.</p><p><strong>Level 2 — Junior full-stack engineer</strong><br>“I understand my codebase, can independently modify it, debug it, work across frontend/API/database, write tests, and deploy it.”</p><p>This is the level I'd make your <strong>immediate target</strong>.</p><p><strong>Level 3 — Strong product engineer</strong><br>“I can take an ambiguous user problem, design the technical/product solution, build it end-to-end, measure whether it works, and maintain it.”</p><p>This is where I’d feel good about you applying aggressively to something like <strong>ChatGPT Education &amp; Learning</strong>, even if you only had a couple years professionally.</p><p><strong>Level 4 — Senior product engineer</strong><br>“I've repeatedly done Level 3 in production at meaningful scale and can make architectural decisions for other engineers.”</p><p>You do <strong>not</strong> need to wait until here before applying to every OpenAI engineering job. Some postings, like Emerging Products, explicitly ask for a senior engineer and clearly belong at this level. (<a href="https://openai.com/careers/full-stack-software-engineer-emerging-products-san-francisco/?utm_source=chatgpt.com" title="Full-Stack Software Engineer, Emerging Products | OpenAI">OpenAI</a>)</p><p>The biggest shift for your GitHub over the next couple years is therefore <strong>depth &gt; repo count</strong>.</p><p>You already have Minecraft, Fourier, CV work, robotics, finance/product concepts, agent experiments, etc. You don't need 40 more repos. You need a few repositories where I can open the README and see:</p><p><strong>problem → user → architecture → implementation → tests → deployment → evaluation → what failed → what you changed.</strong></p><p>And you personally need to be able to open that repo six months later and explain <strong>why every major component exists</strong>, even if Codex helped you write plenty of the code. That is probably the single most important step between your current way of building and representing yourself credibly as a full-stack engineer.</p><p>So I would <strong>not</strong> frame the Education role as “come back in five years.” A better target is: graduate, get into a genuinely technical/product-building role, continue making your portfolio substantially deeper, and reassess in roughly <strong>1–3 years</strong>. OpenAI itself defines 0–3 years as emerging talent, so there is no reason to assume you need a long conventional SWE career before you can take a shot. (<a href="https://openai.com/careers/emerging-talent/?utm_source=chatgpt.com" title="OpenAI">OpenAI</a>)</p></body></html>
</body>
</html>Yeah — the OpenAI roles we were looking at are mostly **not “JPMC analyst program” entry-level roles**.

OpenAI does have true early-career hiring: its Emerging Talent page explicitly says it offers full-time roles in research, applied engineering, and product for people with **0–3 years of experience**, including recent grads and self-taught candidates. ([[OpenAI](https://openai.com/careers/emerging-talent/?utm_source=chatgpt.com)][1]) But the specific **Full-Stack Engineer, ChatGPT Education & Learning** posting is not labeled as an early-career program role. It reads like they want someone who can already operate as an independent product engineer: own features end to end, build across frontend and backend, ship polished experiences, work with React/TypeScript plus Python/Node services, integrate AI capabilities, and handle ambiguity without much scaffolding. ([[OpenAI](https://openai.com/careers/full-stack-engineer-chatgpt-education-and-learning-san-francisco/?utm_source=chatgpt.com)][2])

So I’d think of the comparison like this:

| JPMC analyst-style role                       | OpenAI Education full-stack role                 |
| --------------------------------------------- | ------------------------------------------------ |
| “We’ll train a strong new grad”               | “Show us you can already ship”                   |
| Potential-heavy                               | Evidence-heavy                                   |
| Structured ramp                               | High ownership quickly                           |
| Coursework + internship pedigree matter a lot | Portfolio + production-quality work matter a lot |

For the Education role specifically, I would **not wait until you look like some 10-year senior engineer**. The current posting doesn’t specify a hard years-of-experience requirement. What matters is reaching the point where your GitHub makes someone believe, “This person could own a real feature.” ([[OpenAI](https://openai.com/careers/full-stack-engineer-chatgpt-education-and-learning-san-francisco/?utm_source=chatgpt.com)][2])

For you, that means moving from **lots of interesting prototypes** to roughly **2–4 extremely convincing full-stack projects**. The portfolio should communicate these capabilities:

* You can build a real React/TypeScript frontend rather than just static demos.
* You can design a backend/API in Python or Node and understand the data flow.
* You can persist user state with a database and handle authentication or accounts where appropriate.
* You can integrate an LLM or multimodal model for a reason deeper than “I called an API.”
* You understand loading/error states, retries, validation, basic security, and edge cases.
* You can test meaningful logic and explain architectural decisions.
* At least one project is actually deployed and usable.
* Most importantly for this team, you can demonstrate **product judgment around learning**: what problem does the learner have, why does the interaction work, and how did you evaluate whether it helped?

That last one is where you could build a very differentiated portfolio.

Your current Fourier/softmax-style educational work is pointing in the right direction, but the OpenAI version of that portfolio shouldn't just be:

> “Here is an interactive visualization of softmax.”

It should eventually resemble something like:

> A learner says, “I don’t understand why increasing one logit affects every other probability.” The product identifies the misconception, generates an interactive visualization, lets the learner manipulate logits, adapts the explanation based on their actions, remembers where they struggled, and tests whether they actually understand the concept afterward.

Now you have **frontend + backend + AI + interaction design + learning science + user state + evaluation** in one coherent product. That maps much more directly to what the Learning team says it builds: end-to-end learning experiences, multimodal interfaces, scalable backend services, experimentation, user feedback, and AI-powered educational products. ([[OpenAI](https://openai.com/careers/full-stack-engineer-chatgpt-education-and-learning-san-francisco/?utm_source=chatgpt.com)][2])

### The level I would target

I’d use this mental ladder:

**Level 1 — Prototype builder**
“Given an idea and AI coding tools, I can get something working.”

You’re already around here across multiple projects.

**Level 2 — Junior full-stack engineer**
“I understand my codebase, can independently modify it, debug it, work across frontend/API/database, write tests, and deploy it.”

This is the level I'd make your **immediate target**.

**Level 3 — Strong product engineer**
“I can take an ambiguous user problem, design the technical/product solution, build it end-to-end, measure whether it works, and maintain it.”

This is where I’d feel good about you applying aggressively to something like **ChatGPT Education & Learning**, even if you only had a couple years professionally.

**Level 4 — Senior product engineer**
“I've repeatedly done Level 3 in production at meaningful scale and can make architectural decisions for other engineers.”

You do **not** need to wait until here before applying to every OpenAI engineering job. Some postings, like Emerging Products, explicitly ask for a senior engineer and clearly belong at this level. ([[OpenAI](https://openai.com/careers/full-stack-software-engineer-emerging-products-san-francisco/?utm_source=chatgpt.com)][3])

The biggest shift for your GitHub over the next couple years is therefore **depth > repo count**.

You already have Minecraft, Fourier, CV work, robotics, finance/product concepts, agent experiments, etc. You don't need 40 more repos. You need a few repositories where I can open the README and see:

**problem → user → architecture → implementation → tests → deployment → evaluation → what failed → what you changed.**

And you personally need to be able to open that repo six months later and explain **why every major component exists**, even if Codex helped you write plenty of the code. That is probably the single most important step between your current way of building and representing yourself credibly as a full-stack engineer.

So I would **not** frame the Education role as “come back in five years.” A better target is: graduate, get into a genuinely technical/product-building role, continue making your portfolio substantially deeper, and reassess in roughly **1–3 years**. OpenAI itself defines 0–3 years as emerging talent, so there is no reason to assume you need a long conventional SWE career before you can take a shot. ([[OpenAI](https://openai.com/careers/emerging-talent/?utm_source=chatgpt.com)][1])

[1]: https://openai.com/careers/emerging-talent/?utm_source=chatgpt.com "OpenAI"
[2]: https://openai.com/careers/full-stack-engineer-chatgpt-education-and-learning-san-francisco/?utm_source=chatgpt.com "Full-Stack Engineer, ChatGPT Education & Learning | OpenAI"
[3]: https://openai.com/careers/full-stack-software-engineer-emerging-products-san-francisco/?utm_source=chatgpt.com "Full-Stack Software Engineer, Emerging Products | OpenAI"
